### PR TITLE
Fixes #14: Rework using Mustache instead of dot

### LINKED ISSUE #14: Address DoT vulnerability
https://nodesecurity.io/advisories/798 

1. assess whether this vulnerability may affect end-users of node-request-error-handler (and osprey)
2. find a fix

### PR DESCRIPTION
**Pull request by postatum**
Most of reworked templates are in `lib/messages/json.js `

### CODE REVIEW

**Comment by postatum** [comment]
Reworked a little to make messages callable again so their interface doesn't change for those who use those messages from another libs.

File: lib/index.spec.js [hunk 1]
```diff
@@ -513,7 +513,7 @@ describe('request error handler', function () {
             'application/json': {
               schema: JSON.stringify({
                 type: 'string',
-                enum: ['a']
```
Comment (jstoiko) [inline lib/index.spec.js]: Let's have two separate test cases for enum: one with `['a']` and one with `['a', 'c']`.

File: lib/dot.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+var Mustache = require('mustache')
```
Comment (jstoiko) [inline lib/dot.js]: let's rename `dot.js` and any references to `dot`, no need to keep the name `dot`, something like `template.js` sounds more appropriate

**Review by jstoiko — CHANGES_REQUESTED** [review]

File: lib/index.spec.js (hunk 1 above)
Comment (postatum) [reply lib/index.spec.js]: Done

File: lib/dot.js (hunk 2 above)
Comment (postatum) [reply lib/dot.js]: done

**Review by jstoiko — APPROVED** [review]

**Pull request merged**